mathesar/imports/csv.py

```diff
@@ -13,19 +13,45 @@ def get_csv_reader(csv_file):
     return reader
 
 
-def create_db_table_from_csv(name, schema, csv_reader, engine):
+# TODO: Remove this function once frontend switches to using the API
+def legacy_create_db_table_from_csv(name, schema, csv_reader, engine):
     table = tables.create_string_column_table(
         name, schema, csv_reader.fieldnames, engine,
     )
     return table
 
 
-def create_table_from_csv(name, schema, database_key, csv_file):
+# TODO: Remove this function once frontend switches to using the API
+def legacy_create_table_from_csv(name, schema, database_key, csv_file):
     engine = create_mathesar_engine(database_key)
     csv_reader = get_csv_reader(csv_file)
-    db_table = create_db_table_from_csv(name, schema, csv_reader, engine)
+    db_table = legacy_create_db_table_from_csv(name, schema, csv_reader, engine)
     database = get_database_key(engine)
     schema, _ = Schema.objects.get_or_create(name=db_table.schema, database=database)
     table, _ = Table.objects.get_or_create(name=db_table.name, schema=schema)
     table.create_record_or_records([row for row in csv_reader])
     return table
+
+
+def create_db_table_from_data_file(data_file):
+    engine = create_mathesar_engine(data_file.schema.database)
+    csv_filename = data_file.file.path
+    with open(csv_filename, 'rb') as csv_file:
+        csv_reader = get_csv_reader(csv_file)
+        column_names = csv_reader.fieldnames
+        table = tables.create_string_column_table(
+            name=data_file.file.name.split('/')[-1],
```

Comment:
This will break if we ever support Windows.  I suggest using `os.path.split` here (and using the `os.path` library for file path mangling in general)

Reply:
Thanks for catching this. I haven't used Windows in so long that I completely forgot about different separators.